packages/netlify-cms-core/src/actions/entries.js

```diff
@@ -236,21 +251,44 @@ export function loadLocalBackup() {
   };
 }
 
-export function persistLocalBackup(entry, collection) {
+export function addDraftEntryMediaFile(file) {
+  return { type: ADD_DRAFT_ENTRY_MEDIA_FILE, payload: file };
+}
+
+export function addDraftEntryMediaFiles(files) {
+  return { type: ADD_DRAFT_ENTRY_MEDIA_FILES, payload: files };
+}
+
+export function removeDraftEntryMediaFile(file) {
+  return { type: REMOVE_DRAFT_ENTRY_MEDIA_FILE, payload: file };
+}
+
+export function persistLocalBackup(entry, collection, mediaFiles) {
   return (dispatch, getState) => {
     const state = getState();
     const backend = currentBackend(state.config);
-    return backend.persistLocalDraftBackup(entry, collection);
+
+    // persist any pending related media files and assets
+    const assets = getMediaAssets(state, mediaFiles);
+
+    return backend.persistLocalDraftBackup(entry, collection, mediaFiles, assets);
   };
 }
 
 export function retrieveLocalBackup(collection, slug) {
   return async (dispatch, getState) => {
     const state = getState();
     const backend = currentBackend(state.config);
-    const entry = await backend.getLocalDraftBackup(collection, slug);
+    const { entry, mediaFiles, assets } = await backend.getLocalDraftBackup(collection, slug);
+
     if (entry) {
-      return dispatch(localBackupRetrieved(entry));
+      // load any pending related media files and assets
+      const assetProxies = await Promise.all(
+        assets.map(asset => createAssetProxy(asset.value, asset.fileObj)),
+      );
+      dispatch(addAssets(assetProxies));
```

Comment:
sync assets - `createAssetProxy` creates an `AssetProxy` class (can't used the serialised object directly since we need the class methods).